articles/npm-packages/index.md

```diff
@@ -0,0 +1,194 @@
+
+- Go to the [npmjs website](https://www.npmjs.com/) and click on the **sign up** button.
+
+![npmjs Webpage](/engineering-education/npm-packages/npmjs-homepage.png)
+
+- Enter a username, email address and password.
+
+![Create NPM Account](/engineering-education/npm-packages/npm-signup.png)
+
+- Now, you have to verify your account. Check the inbox of the email address that you provided and click on the verification email. This will verify your npm account.
+
+- You have successfully created an NPM account, and can start publishing packages.
+
+- You need to sign in to your npm account in your console to start publishing packages.
+
+- To do this, open a new folder and start your terminal. Enter the command:
+
+```bash
+npm login
+```
+
+- Now, enter your username, password and email ID.
+
+![NPM Login](/engineering-education/npm-packages/npm-login.png)
+
+- We have successfully signed in.
+
+### Let's Code
+
+- Every package published to the NPM registry should have a **`package.json`** file. Let's start by initializing a project.json file.
+
+- The minimum requirements of a `package.json` file is a name and a version. Let's enter those information into the file.
+
+```json
+{
+  "name": "clist-node",
+  "version": "1.0.0"
+}
+```
+
+- In our sample NPM package, we are going to use the axios package for making API calls. Install it by using this command:
+
+```bash
+npm i axios
+```
+
+_Note: To learn more about Axios, check out its NPM package page [here](https://www.npmjs.com/package/axios)._
+
+- We'll be using the [clist API](https://clist.by/) that returns a list of contests like hackathons, programming contests, etc..,
+
+- Create an account on the [clist website](https://clist.by/api/v1/doc/) and get an API key.
+
+- The entry point of our package would be the **`index.js`** file. Create a file named index.js in the same directory.
+
+```js
+// module.exports exports the function getContests as a promise and exposes it as a module.
+// we can import an exported module by using require().
+module.exports = async function getContests() {
+  const axios = require("axios"); // Importing the Axios module to make API requests
+  var result;
+
+  const username; // Insert Your Username here
+  const api_key; //Insert API key here
+
+  await axios // Making a GET request using axios and requesting information from the API
+    .get(
+      "https://clist.by/api/v1/json/contest/?username=" + username + "&api_key=" + api_key + "&limit=20&end__gt=2020-09-19T00%3A00%3A00"
+    )
+    .then((response) => { // If the GET request is successful, this block is executed
+      return response; // The response of the API call is passed on to the next then block
+    })
+    .then((contests) => { // In this block, we store the response data into a variable 'result'
+      result = contests.data.objects;
+    })
+    .catch((err) => {
+      console.log(err); // Error handler
+    });
+  return result; // The contest data is returned
+};
+```
+
```

Comment:
You should note to the readers that this code requires an API key and is not expected to run straight off the bat because you're just publishing the package.